Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -0,0 +1,880 @@
+/-
+Copyright (c) 2024 María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos Fernández, Xavier Généreux
+-/
+import Mathlib.Algebra.Algebra.Equiv
+import Mathlib.Algebra.Algebra.NonUnitalHom
+import Mathlib.Algebra.BigOperators.Finsupp
+import Mathlib.Algebra.Module.BigOperators
+import Mathlib.LinearAlgebra.Finsupp
+
+/-!
+# Skew Monoid algebras
+This file presents a skewed version of `Mathlib.Algebra.MonoidAlgebra.Basic`.
+
+## Definition
+We define `SkewMonoidAlgebra k G := G →₀ k` attached with a skewed convolution product.
+Here, the product of two elements `f g : SkewMonoidAlgebra k G` is the finitely supported
+function whose value at `a` is the sum of `f x * (x • g y)` over all pairs `x, y`
+such that `x * y = a`.
+
+## Implementation notes
+The associativity of the skewed multiplication depends on the `[MulSemiringAction G k]` instance.
```

Comment:
Is it a mathematical prerequisite or is some instance similar as `MulSemiringAction G k` missing?

Reply:
The proof that we have for associativity of the skew monoid algebra (starting in line 192) uses most of the properties of `MulSemiringAction`, except for `smul_one`. It would be possible to create a class analogous to `MulSemiringAction` that does not require `G` to be a monoid (instead requiring only `Mul G`), and our proof should work for that class, but I don't know if this is worth it (do you know of any interesting examples where G is only a magma?).
Other than that, I don't know a proof that holds in greater generality, but if you do, we would be happy to use it.